realm/realm-library/src/main/java/io/realm/RealmCollection.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.util.Collection;
+import java.util.Date;
+
+/**
+ * {@code RealmCollection} is the root of the collection hierarchy that Realm supports. It defines operations on data
+ * collections and the behavior that they will have in all implementations of {@code RealmCollection}s.
+ *
```

Comment:
[Collections.unmodifiableCollection](http://developer.android.com/reference/java/util/Collections.html#unmodifiableCollection(java.util.Collection<? extends E>)) will not guarantee an immutable collection (since the underlying `RealmResults`/`RealmList` could change/advance_read) , should we warn the user about it here?